drainer/server.go

```diff
@@ -287,6 +292,7 @@ func (s *Server) Start() error {
 		defer func() { go s.Close() }()
```

Comment:
there's race here and 
```
	go func() {
		err2 := <-bs.Errors()
		log.Error("drainer reported error", zap.Error(err2))
		bs.Close()
		os.Exit(2)
	}()
```

both close server and if later one gets called later it may return and exit before the first one finishes.

Reply:
how about just make `Server.Start()` return the the error of `s.syncer.Start()` if it returns error